### DISCUSSION TITLE
How to work with 3D Mesh files and data?

### DISCUSSION THREAD

**Discussion by royvelich** [pyg-team/pytorch_geometric#6962] [Q&A] [answered]
Hi,
I am developing a GNN framework to process 3D mesh files.
So, I have a collection of OBJ files, which I load with the helper function `read_bj` which is provided by the `torch_geometric.io` module. The `data` object that is returned by `read_obj` has two main attributes: `pos` and `face`, while the `edge_index` attribute is missing. Now, what I don't understand, is how to use that `data` object in my convolution layers. Note that the `MessagePassing.propagate` requires an `edge_index` argument, which in m case, is missing.
I would love to hear how do you handle 3D meshes in your scenarios. Is there any example I can follow?

Thanks.

**Comment by rusty1s** [comment, marked as answer]
You can use the `FaceToEdge` transform to convert your `face` representation to a graph.

**Reply by royvelich** [reply]
Thanks.
I also see that `get_mesh_laplacian` returns an `edge_index`. Is it equivalent to what `FaceToEdge` returns?

**Reply by rusty1s** [reply]
Yes, that is indeed similar. It will also give you an `edge_weight` corresponding to the Laplacian.
